static/js/views/new-thing.js

```diff
@@ -373,74 +317,11 @@ class NewThing {
     this.customIconLabel.classList.add('hidden');
     this.customIcon.classList.add('hidden');
 
-    let cls = '';
-    switch (capability) {
-      case 'OnOffSwitch':
-        cls = 'on-off-switch';
-        break;
-      case 'MultiLevelSwitch':
-        cls = 'multi-level-switch';
-        break;
-      case 'ColorControl':
-        cls = 'color-control';
-        break;
-      case 'ColorSensor':
-        cls = 'color-sensor';
-        break;
-      case 'EnergyMonitor':
-        cls = 'energy-monitor';
-        break;
-      case 'BinarySensor':
-        cls = 'binary-sensor';
-        break;
-      case 'MultiLevelSensor':
-        cls = 'multi-level-sensor';
-        break;
-      case 'SmartPlug':
-        cls = 'smart-plug';
-        break;
-      case 'Light':
-        cls = 'light';
-        break;
-      case 'DoorSensor':
-        cls = 'door-sensor';
-        break;
-      case 'MotionSensor':
-        cls = 'motion-sensor';
-        break;
-      case 'LeakSensor':
-        cls = 'leak-sensor';
-        break;
-      case 'PushButton':
-        cls = 'push-button';
-        break;
-      case 'VideoCamera':
-        cls = 'video-camera';
-        break;
-      case 'Camera':
-        cls = 'camera';
-        break;
-      case 'TemperatureSensor':
-        cls = 'temperature-sensor';
-        break;
-      case 'HumiditySensor':
-        cls = 'humidity-sensor';
-        break;
-      case 'Alarm':
-        cls = 'alarm';
-        break;
-      case 'Thermostat':
-        cls = 'thermostat';
-        break;
-      case 'Lock':
-        cls = 'lock';
-        break;
-      case 'Custom':
-        this.customIconLabel.classList.remove('hidden');
-        this.customIcon.classList.remove('hidden');
-        break;
-      default:
-        break;
+    const cls = getClassFromCapability();
```

Comment:
You should probably pass something to the function. 😄

Reply:
Damn, I miss my compiler :sweat_smile: